music21/test/testStream.py

```diff
@@ -284,6 +285,168 @@ def testBeamsMeasure(self):
+        beams = self.get_beams_from_stream(m2)
+
+        self.assertEqual([beam.Beams(),] * 8, beams)
+
+    def test_repeated_1_e_a(self):
+        """
+        Test that the pattern of "1 e a" repeated more than once has correct beams.
+
+        Note:
+            Music21 repr: https://share.getcloudapp.com/jkuyAdrG
+            MuseScore repr (auto-corrected): https://share.getcloudapp.com/12uE7eBA
+        """
+        m = Measure()
+        m.timeSignature = meter.TimeSignature('2/4')
+        for i in range(2):
+            m.append(note.Note(quarterLength=0.25))
+            m.append(note.Note(quarterLength=0.50))
+            m.append(note.Note(quarterLength=0.25))
+
+        m2 = m.makeBeams()
+        beams = self.get_beams_from_stream(m2)
+
+        # Prepare the should be beams
+        first_note_beams = beam.Beams()
+        first_note_beams.append('start')
+        first_note_beams.append('partial', 'right')
+
+        second_note_beams = beam.Beams()
+        second_note_beams.append('continue')
+
+        third_note_beams = beam.Beams()
+        third_note_beams.append('stop')
+        third_note_beams.append('partial', 'left')
+
+        # Now test that they are equal
+        self.assertEqual(first_note_beams, beams[0])
+        self.assertEqual(second_note_beams, beams[1])
+        self.assertEqual(third_note_beams, beams[2])
+
+        self.assertEqual(first_note_beams, beams[3])
+        self.assertEqual(second_note_beams, beams[4])
+        self.assertEqual(third_note_beams, beams[5])
+
+    def test_1_e_n_a(self):
+        """
+        Test that 4 16th notes have proper beams across them all.
+        """
+        m = Measure()
+        m.timeSignature = meter.TimeSignature('1/4')
+        m.repeatAppend(note.Note(quarterLength=0.25), 4)
+
+        m2 = m.makeBeams()
+        beams = self.get_beams_from_stream(m2)
+
+        # Prepare the should be beams
+        first_note_beams = beam.Beams()
+        first_note_beams.append('start')
+        first_note_beams.append('start')
+
+        second_note_beams = beam.Beams()
+        second_note_beams.append('continue')
+        second_note_beams.append('continue')
+
+        third_note_beams = beam.Beams()
+        third_note_beams.append('continue')
+        third_note_beams.append('continue')
+
+        forth_note_beams = beam.Beams()
+        forth_note_beams.append('stop')
+        forth_note_beams.append('stop')
+
+        # Now test that they are equal
+        self.assertEqual(first_note_beams, beams[0])
+        self.assertEqual(second_note_beams, beams[1])
+        self.assertEqual(third_note_beams, beams[2])
+        self.assertEqual(forth_note_beams, beams[3])
+
+    def test_1_e__after_16th_note(self):
+        """
+        Test that a 16th+8th notes after a 16th notes have proper beams.
+
+        Note: MuseScore repr (auto-corrected): https://cl.ly/90ce7b
+        Currently the forth note has 2nd "start" beam which has no "end" anywhere after it.
```

Comment:
correct typo "fourth" for forth -- here and lower.

Reply:
I will think I'll actually remove all the Notes with the screenshots, as all issues has been fixed now. I'll keep a short note with a screenshot of how certain cases should look like